custom-metrics-stackdriver-adapter/pkg/adapter/provider/translator.go

```diff
@@ -76,19 +75,46 @@ func (t *Translator) GetRespForPod(response *stackdriver.ListTimeSeriesResponse,
 		return metricValue, nil
 	}
 	// This code is unreacheable
-	return nil, fmt.Errorf("Illegal state")
+	return nil, apierr.NewInternalError(fmt.Errorf("Illegal state"))
 }
 
-// GetRespForPods returns translates Stackdriver response to a Custom Metric associated
+// GetRespForPods translates Stackdriver response to a Custom Metric associated
 // with multiple pods.
 func (t *Translator) GetRespForPods(response *stackdriver.ListTimeSeriesResponse, podList *v1.PodList, groupResource schema.GroupResource, metricName string, namespace string) (*custom_metrics.MetricValueList, error) {
-	values, err := t.getMetricValuesFromResponse(groupResource, namespace, response)
+	values, err := t.getMetricValuesFromResponse(groupResource, namespace, response, metricName)
 	if err != nil {
 		return nil, err
 	}
 	return t.metricsFor(values, groupResource, metricName, podList)
 }
 
+// ListMetricDescriptors returns Stackdriver request for all custom metrics descriptors.
+func (t *Translator) ListMetricDescriptors() *stackdriver.ProjectsMetricDescriptorsListCall {
+	return t.service.Projects.MetricDescriptors.List(fmt.Sprintf("projects/%s", t.config.Project)).
+		Filter(joinFilters(t.filterForMetricPrefix(), t.filterForCluster(), t.filterForAnyObject()))
+}
+
+// GetMetricsFromSDDescriptorsResp returns an array of MetricInfo for all metric descriptors
+// returned by Stackdriver API that satisfy the requirements:
+// - metricKind is "GAUGE"
+// - valueType is "INT64" or "DOUBLE"
```

Comment:
Do we support both in the API?

Reply:
Custom Metrics API uses `resource.Quantity` and doesn't say anything about underlying SD API. I think it's up to us what we support and the most obvious decision seems to be to allow any numeric value.

Reply:
SGTM